src/Nethermind/Nethermind.Merkleization/Merkleizer.cs

```diff
@@ -574,6 +723,20 @@ public void Feed(IReadOnlyList<Bytes32> value, ulong maxLength)
         Feed(_chunks[^1]);
     }
 
+    public void Feed(IReadOnlyList<ulong> value, ulong maxLength)
+    {
+        // TODO: If UInt256 is the correct memory layout
+        UInt256[] subRoots = new UInt256[value.Count];
```

Comment:
ArrayPoolList?

Reply:
ArrayPoolList seems a bit too complex for such task. Mind we add something like 
```
public struct ArrayPoolSpan<T>(ArrayPool<T> arrayPool, int length) : IDisposable
{
    private readonly T[] _array = arrayPool.Rent(length);
    private readonly int _length = length;

    public ArrayPoolSpan(int length) : this(ArrayPool<T>.Shared, length) { }

    public readonly ref T this[int index] => ref _array[index];

    public static implicit operator Span<T>(ArrayPoolSpan<T> self) => self._array.AsSpan(0, self._length);
    public static implicit operator ReadOnlySpan<T>(ArrayPoolSpan<T> self) => self._array.AsSpan(0, self._length);

    public readonly void Dispose() => arrayPool.Return(_array);
}
```